includes/tasks/class-wc-calypso-task-add-domain.php

```diff
@@ -67,8 +67,23 @@ public function get_time() {
 	public function get_action_url() {
 		$status      = new \Automattic\Jetpack\Status();
 		$site_suffix = $status->get_site_suffix();
+		$domain_path = sprintf( "https://wordpress.com/domains/add/%s", $site_suffix );
+		$home_url    = \home_url( '', 'https' );
 
-		return sprintf( "https://wordpress.com/domains/add/%s", $site_suffix );
+		if ( ! \str_ends_with( $home_url, '.wpcomstaging.com' ) ) {
+			return $domain_path;
+		}
+
+		if ( ! \str_starts_with( $home_url, 'https://woo-' ) && ! \str_starts_with( $home_url, 'https://wooexpress-' ) ) {
+			return $domain_path;
+		}
+
+		$blog_name = \get_option( 'blogname' );
+		if ( empty( $blog_name ) ) {
+			return $domain_path;
+		}
+
+		return sprintf( '%s?suggestion=%s', $domain_path, rawurlencode( $blog_name ) );
```

Comment:
Hi @daledupreez  Since we don't limit user input format in NUX, `$blog_name` might contain non-alphanumeric, upper characters, or would be too long. Do you think we need to normalize `rawurlencode( $blog_name )` further? For example, convert it to lowercase, and replace blank spaces with `-`.

Reply:
Good thought, @chihsuan!

The code responsible for performing domain searches should be able to handle whatever we throw at it, as users can also enter whatever they like in the input field. All we are doing with this PR is supplying specific content for the input field in the UI, so I don't think we need to be more defensive than this.

Reply:
> The code responsible for performing domain searches should be able to handle whatever we throw at it

Nice, that’s good to hear! :slightly_smiling_face: